### PR TITLE
Use Timeout.timeout for TCP connection

### PR DESCRIPTION
**Pull request by zanker**
Unfortunately, getting a pure solution to timeouts is essentially impossible without rewriting the entire stack yourself. Which I might actually do for fun later, but for the time being. This is the simplest way of getting a connect timeout :(

### CODE REVIEW

**Comment by zanker** [comment]
@httprb/owners fyi if someone can :+1: quickly

File: lib/http/timeout/per_operation.rb [hunk 1]
```diff
@@ -20,40 +17,22 @@ def initialize(*args)
...
 
-        case addr
-        when Resolv::IPv4
-          family = Socket::AF_INET
-        when Resolv::IPv6
-          family = Socket::AF_INET6
-        else fail ArgumentError, "unsupported address class: #{addr.class}"
+      def connect_ssl
+        socket.connect_nonblock
+      rescue IO::WaitReadable
```
Comment (tarcieri) [inline lib/http/timeout/per_operation.rb]: I guess I missed this on the last pass, but as the SSL/TLS handshake makes many roundtrips, I believe it's possible for IO::WaitWritable to be raised here too.

**Comment by tarcieri** [comment]
This looks fine to me

File: lib/http/timeout/per_operation.rb (hunk 1 above)
Comment (zanker) [inline lib/http/timeout/per_operation.rb]: Oops I broke Excon then!!

**Comment by ixti** [comment]
:+1: from me as well :D

**Pull request merged**